src/components/svelte-web.ts

```diff
@@ -0,0 +1,226 @@
+        // so we can recreate it as faithfully as possible.
+        // Note: We might be able to do some additional hackery here
+        // to copy over even more information from $$.ctx and exactly
+        // maintain the component state!
+        const existingProps = Object.keys(this.component?.$$.props ?? {})
+          .map((k) => [k, this[k]])
+          .reduce((prev, [key, value]) => ({ ...prev, [key]: value }), {})
+
+        // If the component already exists, destroy it. This is,
+        // unfortunately, necessary as there is no way to update slotted
+        // content in the output Svelte compiles to. This is a problem
+        // even when not doing crazy things:
+        // https://github.com/sveltejs/svelte/issues/5312
+        if (this.component) {
+          this.component.$destroy()
+        }
+
+        // Finally, we actually create the component
+        this.component = new component({
+          // Target this shadowDOM, so we get nicely encapsulated
+          // styles
+          target: shadow,
+          props: {
+            // Copy over existing props (there might be none, if
+            // this is our first render).
+            ...existingProps,
+            // Create WebComponent slots for each Svelte slot we
+            // have content for. This has to be done at render or
+            // Svelte won't support fallback content.
+            $$slots: slots,
+            // Not sure what this is needed for but Svelte crashes
+            // without it. I think this might be related to slot
+            // props:
+            // https://svelte.dev/tutorial/slot-props
+            $$scope: { ctx: [] }
+          }
+        })
+      }
+
+      // Unfortunately we need a DOMMutationObserver to let us know when
+      // slotted content changes because we dynamically create & remove
+      // slots. This is for two reasons:
+      // 1) At runtime, we don't know what slots our Svelte component has
+      // 2) Even if we did, if we generated all of the slots at mount time
+      //    then Svelte would never render any of the fallback content,
+      //    event if the slot was empty.
+      new MutationObserver(updateSlots).observe(this, {
+        childList: true,
+        attributes: false,
+        attributeOldValue: false,
+        subtree: false,
+        characterData: false,
+        characterDataOldValue: false
+      })
+
+      // Update slots on create.
+      updateSlots()
+
+      // For some reason setting this on |SvelteWrapper| doesn't work properly.
+      for (const prop of props) {
+        Object.defineProperty(this, prop, {
+          enumerable: true,
+          get() {
+            const contextIndex = c.$$.props[prop]
+            return this.component.$$.ctx[contextIndex]
+          },
+          set(value) {
+            if (reflectToAttributes.has(typeof value)) {
+              this.setAttribute(prop, value)
+            }
+            this.component.$$set({ [prop]: value })
```

Comment:
Why `.$$set` instead of [`.$set` as documented here](https://svelte.dev/docs#run-time-client-side-component-api-$set)?

Reply:
Because when I did `Object.entries` on a Svelte component `$$set` showed up and `$set` did not :laughing:. I've swapped to using this in a new PR, as it definitely seems like the (more) blessed path.